homeassistant/components/shelly/__init__.py

```diff
@@ -82,19 +90,29 @@ def __init__(self, hass, entry, device: aioshelly.Device):
             hass,
             _LOGGER,
             name=device.settings["name"] or device.settings["device"]["hostname"],
-            update_interval=timedelta(seconds=5),
         )
         self.hass = hass
         self.entry = entry
         self.device = device
+        self._update_from_multicast = False
+
+        self.device.subscribe_updates(self._multicast_update)
+
+    def _multicast_update(self, data):
+        """Handle multicast device update."""
+        self._update_from_multicast = True
+        self.hass.async_create_task(self.async_refresh())
```

Comment:
This makes no sense. Why even call refresh? We should instead just call `self.async_write_ha_state()` as the new data is already available.

Reply:
Oh, this is in the device wrapper 🤔

Reply:
We should update the UpdateCoordinator to be able to push results in. I will give that a stab.

Reply:
Updated to use "async_set_updated_data"